src/Factory.php

```diff
@@ -127,31 +119,37 @@ protected function buildClaims()
 
         // add the default claims
         foreach ($this->defaultClaims as $claim) {
-            $this->addClaim($claim, $this->$claim());
+            $this->addClaim($claim, $this->claimFactory->make($claim));
         }
 
         // add custom claims on top, allowing them to overwrite defaults
         $this->addClaims($this->getCustomClaims());
```

Comment:
Feature: have custom claims use the claim factory as well.

The custom claims in this factory don't necessarily mean "custom" in the sense of the `Custom` claim. The user might still have defined a special claim class with validation and behavior. So in that sense they're more like "additional" claims beyond the default set, and it would be nice if there were a way to still run through their classes' constructors/methods.

Additionally, if you support moving the defaults into the claims, you could allow unkeyed array entries that are just claim names. So this would work:
```php
Factory::customClaims(['name' => 'myApplication', 'cnf'])->make();
// if those exist in the class map, would result in claim Factory running:
$classMap['name']->setValue('myApplication');
$classMap['cnf']->setDefault();
```
(Although technically detecting that probably requires removing support for numerically-named claims.)